test/lib/tcpReceiver.test.js

```diff
@@ -0,0 +1,271 @@
+    conn.on('close', (hadError) => {
+      if (!done && hadError) {
+        callback(hadError);
+      }
+    });
+
+    conn.on('data', (chunk) => {
+      replyString += chunk.toString();
+      const replies = replyString.split('\n');
+      if (replies.length > taskCount) {
+        for (let i = 0; i < taskCount; i++) {
+          replies[i] = JSON.parse(replies[i]);
+        }
+      }
+
+      // close the connection to not leak the file descriptor.  Ignore 'close' events.
+      // FIXME:  closing the connection interferes with subsequent connections too ??
+
+      conn.end();
+      done = true;
+
+      if (taskCount === 1) {
+        callback(null, replies[0]);
+      } else {
+        callback(null, replies.slice(0, taskCount));
+      }
+    });
+
+    if (flags.alreadyJson) {
+      let batch = '';
+      for (const task of tasks) {
+        batch += task;
+        if (task[task.length - 1] !== '\n') {
+          batch += '\n';
+        }
+      }
+      conn.write(batch);
+    } else {
+      let batch = '';
+      for (const task of tasks) {
+        batch += `${JSON.stringify(task)}\n`;
+      }
+      conn.write(batch);
+    }
+  });
+}
+
+function startReceiver(taskReceivedCallback, callback, options) {
+  if (!taskReceivedCallback) {
+    taskReceivedCallback = () => null;
+  }
+
+  if (!options) {
+    options = {
+      type: 'tcp'
+    };
+  }
+
+  runner.start(options, taskReceivedCallback, () => {
+    setTimeout(callback, 20);
+  });
+}
+
+function stopReceiver() {
+  runner.stop();
+}
+
+describe('tcp receiver', () => {
+  function makeTasks(count) {
+    const tasks = [];
+
+    for (let i = 0; i < count; i++) {
+      const id = uniqid();
+      const task = makeCopy(sampleTask);
+      task.targetFunction = 'run';
+      task.requestId = id;
+      task.response = makeCopy(task.response);
+      task.response.headers = makeCopy(task.response.headers);
+      task.response.headers['x-test-requestid'] = id;
+      task.blScript = "function run(req, res, modules) {res.body = {id: '#{id}'}; res.complete(201); return '#{id}';}";
+      tasks.push(task);
+    }
+
+    return tasks;
+  }
+
+  function checkTaskReplies(tasks, replies) {
+    for (const task of tasks) {
+      should.exist(replies[task.taskId]);
+    }
+  }
+
+  afterEach((done) => {
+    stopReceiver();
+    done();
+  });
+
+  it('should do a healthcheck', (done) => {
+    startReceiver(null, () => {
+      sendToRunner({ healthCheck: 1 }, (err, obj) => {
+        if (err) {
+          return done(err);
+        }
+
+        obj.status = 'ready';
+        return done();
+      });
+    });
+  });
+
+  it('should reject invalid json', (done) => {
+    startReceiver(null, () => {
+      sendToRunner('invalid json', true, (err, obj) => {
+        if (err) {
+          return done(err);
+        }
+        obj.isError.should.be.true();
+        obj.debugMessage.should.containEql('unable to parse');
+        return done();
```

Comment:
So the linter is telling me that those should be there for consistency.  The argument for this is that in general, a function should always behave the same - it should either return something or it shouldn't.  (In coffeescript, we had implicit return, so *everything* technically returned something.  

Here, since one codepath returns `done(err)`, then the second codepath should also 'return` something, so the function behaves consistently.  The other option here would be to do:

```
if (err) {
  done(err)};
} else {
  // code
  done();
}
```

Then we have consistency - no return from this function.